addons/sale_project/models/project.py

```diff
@@ -119,7 +120,9 @@ def _compute_display_sales_stat_buttons(self):
 
     def action_view_sols(self):
         self.ensure_one()
-        all_sale_order_lines = self._fetch_sale_order_items({'project.task': [('state', 'in', self.env['project.task'].OPEN_STATES)]})
+
+        # Use sudo to avoid AccessErrors when the SOLs belong to different companies.
+        all_sale_order_lines = self._fetch_sale_order_items({'project.task': [('state', 'in', self.env['project.task'].OPEN_STATES)]}).sudo()
```

Comment:
Sorry but I don't see why we need the sudo here since we will just need the ids with that variable. 🤔 

Is it really needed?

Reply:
As you can see from the above screenshot, viewing the SOLs results in access errors so `action_view_sols` must be modified. At first I thought the fix was to just use `sudo`, but now I've realized that the function just returns the IDs, which will be queried later I suppose. So I've modified it to only return the IDs of SOLs that belong to the current company.

Reply:
The problem by adding the company inside the domain is the user we will see the SO inside the other company even if he is in multi company mode.

I will try to reproduce the issue to see exactly where is the issue. 🤔

Reply:
@xavierbol The issue is that SOL IDs from company A shouldn't be returned when current company is company B. When the SOL is used to compute something else (like project's `sale_order_line_count` for instance), `sudo` must be used.